src-graphql/client.js

```diff
@@ -22,15 +22,59 @@ export default class Client {
     });
   }
 
+  fetchAll(paginatedModels) {
+    return this.graphQLClient.fetchNextPage(paginatedModels).then(({model}) => {
+      // Until we know how hasNextPage will be exposed, we query until the result is empty
+      if (model.length === 0) {
+        return paginatedModels;
+      }
+
+      paginatedModels.push(...model);
+
+      return this.fetchAll(paginatedModels);
+    });
+  }
+
   fetchAllProducts(query = productConnectionQuery()) {
-    return this.graphQLClient.send(query(this.graphQLClient)).then((response) => {
-      return response.model.shop.products;
+    return this.graphQLClient.send(query(this.graphQLClient)).then(({model, data}) => {
+      const promises = model.shop.products.reduce((promiseAcc, product, i) => {
+
+        // Fetch the rest of the images for this product
+        if (data.shop.products.edges[i].node.images && data.shop.products.edges[i].node.images.pageInfo.hasNextPage) {
+          promiseAcc.push(this.fetchAll(product.images));
+        }
+
+        // Fetch the rest of the variants for this product
+        if (data.shop.products.edges[i].node.variants && data.shop.products.edges[i].node.variants.pageInfo.hasNextPage) {
```

Comment:
Can you make a const to reference `data.shop.products.edges[i].node` throughout this function?

It will just make it more readable